numba/cuda/dispatcher.py

```diff
@@ -230,11 +396,13 @@ def max_cooperative_grid_blocks(self, blockdim, dynsmemsize=0):
         return active_per_sm * sm_count
 
     def launch(self, args, griddim, blockdim, stream=0, sharedmem=0):
+        from .cudadrv import enums
         # Prepare kernel
         cufunc = self._codelibrary.get_cufunc()
```

Comment:
Is the `cufunc` sufficiently invariant for it to be cached and/or baked into the launcher?

Reply:
It's invariant for the duration of a single process, but can't be cached on-disk as it will be different each time the module is loaded to the device.

Reply:
(Also. it's not invariant across devices)